addons/im_livechat/static/src/embed/core/thread_service_patch.js

```diff
@@ -37,55 +36,54 @@ patch(ThreadService.prototype, {
         this.chatWindowService = services["mail.chat_window"];
         this.chatbotService = services["im_livechat.chatbot"];
         this.notification = services.notification;
+        /** @type {Promise<import("@mail/core/common/thread_model").Thread>?} */
+        this.persistPromise = null;
     },
 
-    getMessagePostRoute(thread) {
-        if (thread.type !== "livechat") {
-            return super.getMessagePostRoute(...arguments);
+    /**
+     * Persist the given thread  and swap it with the temporary thread.
+     *
+     * @param {import("@mail/core/common/thread_model").Thread} thread
+     * @returns {import("@mail/core/common/thread_model").Thread} The
+     * persisted thread.
+     */
+    async persistThread(thread) {
+        if (thread.id !== this.TEMPORARY_ID) {
+            return thread;
         }
-        return "/im_livechat/chat_post";
-    },
-
-    async getMessagePostParams({ thread, body }) {
-        if (thread.type !== "livechat") {
-            return super.getMessagePostParams(...arguments);
+        if (this.persistPromise) {
+            return this.persistPromise;
+        }
+        this.persistPromise = this.getLivechatThread({ persisted: true });
+        let persistedThread;
+        try {
+            persistedThread = await this.persistPromise;
+        } finally {
+            this.persistPromise = null;
+        }
+        const chatWindow = this.store.chatWindows.find((c) => c.threadLocalId === thread.localId);
+        if (!persistedThread) {
+            this.chatWindowService.close(chatWindow);
+            this.remove(thread);
+            return;
         }
-        return {
-            uuid: thread.uuid,
-            message_content: await prettifyMessageContent(body),
-        };
+        chatWindow.thread = persistedThread;
+        this.remove(thread);
+        if (this.chatbotService.active) {
+            await this.chatbotService.postWelcomeSteps();
+        }
+        return persistedThread;
```

Comment:
`getLivechatThread` has no reason to be present in `thread_service`. This is a `livechat_service` concern.
Also `persistThread` is a livechat concern, so should become a `persistLivechat` in `livechat_service`.

Reply:
Initially this was done this way because the thread service depends on the livechat thus the livechat service cannot depend on the thread one to avoid cyclic dependencies. If we agree to use `this.env.services` regardless of dependencies, this could be improved. I've already started to work on it on another branch (along with `get/_create{session}` so I'd rather to keep it for a follow up of this PR.

Reply:
Let's do that in follow-up PR them. We want to replace `mail.thread` service so there's no so many patches and bad function split. That's why it'd nice to not make it worst by adding new functions there ;)